compiler/cpp/src/thrift/thriftl.ll

```diff
@@ -248,6 +250,18 @@ literal_begin (['\"])
 }
 "&"                  { return tok_reference;            }
 
+{uuidconstant} {
+  /* Placed before intconstant, because uuids start with either "{" or directly with some hex digit */
+  yylval.uconst = strdup(yytext);
+  return tok_uuid_constant;
+}
+
+{guidconstant} {
+  yylval.uconst = strdup(&yytext[1]);
+  yylval.uconst[36] = '\000';
+  return tok_uuid_constant;
+}
```

Comment:
oh I see you stripped the curly braces from the token so language libraries do not need to support the parsing of this form.